src/main/java/net/starlark/java/eval/StarlarkInt.java

```diff
@@ -505,10 +505,32 @@ public static StarlarkInt subtract(StarlarkInt x, StarlarkInt y) {
 
   /** Returns x * y. */
   public static StarlarkInt multiply(StarlarkInt x, StarlarkInt y) {
-    if (x instanceof Int32 && y instanceof Int32) {
-      long xl = ((Int32) x).v;
-      long yl = ((Int32) y).v;
-      return StarlarkInt.of(xl * yl);
+    try {
+      long xl = x.toLongFast();
+      long yl = y.toLongFast();
+
+      // Copy-paste from Guava LongMath.saturatedMultiply https://git.io/JIlZP
```

Comment:
We can't accept code copied from other projects. Please internalize the algorithm and write a new implementation of the algorithm.